Dockerfile

```diff
@@ -56,40 +56,68 @@ RUN apt-get update && \
   apt-get clean -y && \
   rm -rf /var/lib/apt/lists/*
 
+# update NVIDIA keys
+# see https://github.com/NVIDIA/nvidia-docker/issues/1632
+RUN rm /etc/apt/sources.list.d/cuda.list && \
+    rm /etc/apt/sources.list.d/nvidia-ml.list && \
+    apt-key del 7fa2af80 && \
+    wget https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64/cuda-keyring_1.0-1_all.deb && \
+    dpkg -i cuda-keyring_1.0-1_all.deb
+
+
+# Build packages
+RUN apt-get update -q && \
+    apt-get install -y -q \
+        build-essential \
+        vim \
+        openssh-client \
+        libgl1-mesa-dev \
+        libxt-dev
+
+
+# Generate ssh key to access private repo
+RUN ssh-keygen -q -t ed25519 -C 'danlipsa@danesfield-conda-build' -N '' -f /root/.ssh/id_ed25519 && \
+    echo "Copy the following key to gitlab, Preferences, SSH Keys:" && \
+    cat /root/.ssh/id_ed25519.pub && \
+    ssh-keyscan -t rsa gitlab.kitware.com >> ~/.ssh/known_hosts && \
+    sleep 30
```

Comment:
I don't think this is the way to do this (not to mention it may leave around your generated SSH key in the built image layers).  While I haven't done this myself, there's a Docker supported way to pass in your SSH key without needing to do this (and without leaving anything behind), see: https://docs.docker.com/develop/develop-images/build_enhancements/#using-ssh-to-access-private-data-in-builds

Alternatively, you could have a step / script outside of the Dockerfile that pulls down the private repos, then you just copy them into the container during the build.

Reply:
@dmjoy I like the docker supported way to use the ssh key. I ended up not using this Dockerfile to build but used tiler.dockerfile instead. @jacobderosa got this to build correctly by running different scripts in different environment. Maybe Jacob can replace this as well.